lib/verbose/output.js

```diff
@@ -0,0 +1,68 @@
+import {inspect} from 'node:util';
+import stripFinalNewline from 'strip-final-newline';
+import {escapeLines} from '../arguments/escape.js';
+import {PIPED_STDIO_VALUES} from '../stdio/forward.js';
+import {verboseLog} from './log.js';
+
+export const handleStreamsVerbose = ({stdioStreams, options, isSync, stdioState, verboseInfo}) => {
+	if (!shouldLogOutput(stdioStreams, options, isSync, verboseInfo)) {
+		return stdioStreams;
+	}
+
+	const [{fdNumber}] = stdioStreams;
+	return [...stdioStreams, {
+		...stdioStreams[0],
+		type: 'generator',
+		value: verboseGenerator.bind(undefined, {stdioState, fdNumber, verboseInfo}),
+	}];
+};
+
+// `ignore` opts-out of `verbose` for a specific stream.
+// `ipc` cannot use piping.
+// `inherit` would result in double printing.
+// They can also lead to double printing when passing file descriptor integers or `process.std*`.
+// This only leaves with `pipe` and `overlapped`.
+const shouldLogOutput = (stdioStreams, {encoding}, isSync, {verbose}) => verbose === 'full'
+	&& !isSync
```

Comment:
`execaSync()` cannot currently use this because it cannot use transforms and this feature uses transforms.

However, I am planning on adding support for both to `execaSync()`.